python/lsst/drp/tasks/assemble_coadd.py

```diff
@@ -1105,25 +1105,40 @@ def _setValidPolygons(self, warpList, psfMatchedWarpRefList):
 
         Raises
         ------
-        AssertionError
-            Raised if the detector IDs in the coaddInputs do not match.
+        ValueError
+            Raised if PSF-matched warps have detectors that are absent in the
+            (direct) warps.
         """
         for warp, psfMatchedWarpRef in zip(warpList, psfMatchedWarpRefList):
             psfMatchedCcdTable = psfMatchedWarpRef.get(component="coaddInputs").ccds
             ccdTable = warp.getInfo().getCoaddInputs().ccds
-            psfMatchedCcdTable.sort()
-            ccdTable.sort()
-            for idx in range(len(ccdTable)):
-                assert ccdTable[idx].id == psfMatchedCcdTable[idx].id, "ID mismatch"
-                if not psfMatchedCcdTable[idx].validPolygon:
+
+            # In some (literal) edge cases, a small part of the CCD may be
+            # present in directWarp that gets excluded in psfMatchedWarp. It is
+            # okay to leave validPolygon for those CCDs empty. However, the
+            # converse is not expected, and raise an error if that is the case.
```

Comment:
"raise" -> "will raise" or "raises"

(I normally wouldn't comment about something so small, but since you're going to touch this file again, I thought I would).

Reply:
I had meant `raise` as an imperative to the developer :-D ... and I don't think I'd follow the logic of what is going down here if I revisited a year from now, so I figured I'll add a comment.

Reply:
I think the "and" makes this a bit of a garden path sentence. How about this? 

> However, the converse is not expected; raise an error if that is the case.

Reply:
I just changed this to be in passive.